light/store/db/db.go

```diff
@@ -289,32 +277,28 @@ func (s *dbs) Size() uint16 {
 	return s.size
 }
 
-func (s *dbs) lbKey(height int64) []byte {
-	return []byte(fmt.Sprintf("lb/%s/%020d", s.prefix, height))
+func (s *dbs) sizeKey() []byte {
+	// heights always starts from 1. Hence in order to avoid
+	// key collisions we reserve size to be the key at height 0.
+	// note this already will contain the specific light client
+	// prefix.
+	return s.lbKey(0)
 }
 
-var keyPattern = regexp.MustCompile(`^(lb)/([^/]*)/([0-9]+)$`)
-
-func parseKey(key []byte) (part string, prefix string, height int64, ok bool) {
-	submatch := keyPattern.FindSubmatch(key)
-	if submatch == nil {
-		return "", "", 0, false
-	}
-	part = string(submatch[1])
-	prefix = string(submatch[2])
-	height, err := strconv.ParseInt(string(submatch[3]), 10, 64)
+func (s *dbs) lbKey(height int64) []byte {
+	buf := make([]byte, 0)
+	key, err := orderedcode.Append(buf, s.prefix, height)
 	if err != nil {
-		return "", "", 0, false
+		panic(err)
 	}
-	ok = true // good!
-	return
+	return key
 }
 
-func parseLbKey(key []byte) (prefix string, height int64, ok bool) {
-	var part string
-	part, prefix, height, ok = parseKey(key)
-	if part != "lb" {
-		return "", 0, false
+func parseHeightFromKey(key []byte) (height int64) {
+	var prefix string
+	_, err := orderedcode.Parse(string(key), &prefix, &height)
+	if err != nil {
+		panic(err)
```

Comment:
what does panicking in this context mean? that someone meddled with storage bytes?

Reply:
It means that the given key doesn't follow the structure string/int64 format created by the orderedcode library. I can't exactly think of a circumstance where that will happen because we are always iterating over the prefix/height keys anyway so I'm not sure how someone could inject a key/value pair inbetween that the iterator would get stuck on.